registry/storage/driver/neofs/fileWriter.go

```diff
@@ -0,0 +1,453 @@
+package neofs
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"io"
+
+	storagedriver "github.com/distribution/distribution/v3/registry/storage/driver"
+	"github.com/distribution/distribution/v3/registry/storage/driver/neofs/transformer"
+	"github.com/nspcc-dev/neofs-sdk-go/checksum"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/tzhash/tz"
+)
+
+type payloadChecksumHasher struct {
+	hasher         hash.Hash
+	checksumWriter func()
+}
+
+type writer struct {
+	ctx    context.Context
+	driver *driver
+	path   string
+
+	closed    bool
+	committed bool
+	cancelled bool
+
+	maxSize        uint64
+	buffer         []byte
+	written        uint64
+	splitInfo      *object.SplitInfo
+	previous       []*oid.ID
+	chunkWriter    io.Writer
+	targetInit     func() transformer.ObjectTarget
+	target         transformer.ObjectTarget
+	current        *object.RawObject
+	parent         *object.RawObject
+	currentHashers []*payloadChecksumHasher
+	parentHashers  []*payloadChecksumHasher
+}
+
+const tzChecksumSize = 64
```

Comment:
What does  "tz" mean?

Reply:
TZ is a Tillich-Zemor checksum type. https://github.com/nspcc-dev/tzhash